src/functional/ffront/field_operator_ast.py

```diff
@@ -176,6 +159,6 @@ class Return(Stmt):
 
 class FieldOperator(LocatedNode, SymbolTableTrait):
     id: SymbolName  # noqa: A003
-    params: list[DataSymbol]
+    params: list[Symbol[common_types.DataType]]
```

Comment:
This works but it is probably more correct to use a `DataSymbolT` `TypeVar` bound to `common_types.DataType`

Reply:
You are absolutely right (would be great if this would throw an error).

Reply:
Fixed. I've changed `FieldOperator.{params, closure}` and `ExternalImport.symbols` to contain `DataSymbols` only.

Reply:
Fixed. We also discussed offline to change closure and external import to only allow `DataType`, but I recognized this is not sufficient as we also need `OffsetType`.